samples: matter: Enable build with the factory data support. (#11980)
nrfconnect/sdk-nrf: samples/matter/lock/README.rst

See `Testing door lock using Bluetooth LE with Nordic UART Service`_ for more information about how to test this feature.

Factory data support
====================

.. matter_door_lock_sample_factory_data_start

> In this example factory data support is enabled by default for each build types except ``no_dfu``.
> This means that a new factory data set will be generated when building for the target board automatically.
> 
> To disable factory data support set to ``n`` the following configs:
> 
>    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA`
>    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD`
> 
> For example, to build the example without factory data support for ``nrf52840dk_nrf52840`` target, run the following command:
> 
> .. code-block:: console
> 
>    west build -b nrf52840dk_nrf52840 -- -DCONFIG_CHIP_FACTORY_DATA=n -DCONFIG_CHIP_FACTORY_DATA_BUILD=n
> 
> To learn more about factory data, read the :doc:`matter:nrfconnect_factory_data_configuration` page in the Matter documentation.

Reviewer: ```suggestion
In this sample, the factory data support is enabled by default for all build types except for the ``no_dfu`` build type.
This means that a new factory data set will be automatically generated when building for the target board.

To disable factory data support, set the following Kconfig options to ``n``:

   * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA`
   * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD`

To learn more about factory data, read the :doc:`matter:nrfconnect_factory_data_configuration` page in the Matter documentation.
```
I think the example should not be needed. How to build is explained in other places, and the Kconfigs can be set to ``n`` in different places.

       - MT:8IXS142C00KA0648G00
       - 34970112332

.. matter_door_lock_sample_onboarding_start

> When the factory data support is enabled the onboarding information will be stored in a build directory.
> To obtain the onboarding codes go to the build directory (for example example_dir/build/zephyr) and look for the following files:
> 
>    * ``factory_data.png`` - an image that contains generated QR Code
>    * ``factory_data.txt`` - a text file that contains QR Code Payload and Manual pairing code

Reviewer: ```suggestion
When the factory data support is enabled, the onboarding information will be stored in the build directory in the following files:

   * The :file:`factory_data.png` file includes the generated QR code.
   * The :file:`factory_data.txt` file includes the QR Code Payload and the manual pairing code.
```
